go/database/mpt/archive_trie.go

```diff
@@ -328,6 +328,19 @@ func (a *ArchiveTrie) Flush() error {
 	)
 }
 
+func (a *ArchiveTrie) VisitTrie(block uint64, visitor NodeVisitor) error {
+	a.rootsMutex.Lock()
+	if uint64(a.roots.length()) < block {
+		a.rootsMutex.Unlock()
+		return fmt.Errorf("block %d is not present", block)
+
+	}
+
+	ref := a.roots.get(block).NodeRef
+	a.rootsMutex.Unlock()
+	return a.forest.VisitTrie(&ref, visitor)
```

Comment:
```suggestion
	view, err := a.getView(block)
	if err != nil {
		return err
	}
	return view.VisitTrie(visitor)
```

Reply:
getting the view this way is more compact but also already checks for errors, which would be otherwise missing.